snapcraft/internal/project_loader/_extensions/_desktop_common.py

```diff
@@ -0,0 +1,79 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any, Dict
+
+from ._extension import Extension
+
+
+class DesktopCommonExtension(Extension):
+    """The Common Desktop extension.
+    This extension is to be subclassed by the gnome and kde extensions.
```

Comment:
If it's sole purpose is being subclassed, can we please make sure that people cannot use this by hand?  Otherwise this will spawn new unanticipated scenarios that need to be supported forever.

Reply:
the underscore in the file name should prevent snapcraft from calling it directly. That is what I understood from reading the extension loading code, at least. Perhaps @sergiusens or @cmatsuoka can confirm?

Reply:
It's true that the base desktop-common extension can't be loaded, but it fails in a rather ugly way because the class name doesn't match the expected extension module name. I'll propose a minor patch to handle it.